src/Features/Core/Diagnostics/LiveUpdateArgsId.cs

```diff
@@ -5,18 +5,32 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// <summary>
     /// Base type of a type that is used as <see cref="DiagnosticsUpdatedArgs.Id"/> for live diagnostic
     /// </summary>
-    internal class UpdateArgsId
+    internal class LiveUpdateArgsId : ErrorSourceId
     {
         public readonly DiagnosticAnalyzer Analyzer;
 
-        protected UpdateArgsId(DiagnosticAnalyzer analyzer)
+        protected LiveUpdateArgsId(DiagnosticAnalyzer analyzer)
         {
             Analyzer = analyzer;
         }
 
+        public override string ErrorSource
+        {
+            get
+            {
+                if (Analyzer == null)
+                {
+                    return string.Empty;
+                }
+
+                var type = Analyzer.GetType();
+                return $"{type.Name} [{type.Assembly.GetName().Name}]";
```

Comment:
Should probably use type.FullName to disambiguate multiple analyzers with same type name but different namespaces in same assembly.